#ifndef LINPLAYER_H
#define LINPLAYER_H

#include <QtWidgets/QMainWindow>
#include <QFileDialog>
#include "ui_linplayer.h"
#include "PlayCore.h"



class LinPlayer : public QMainWindow
{
	Q_OBJECT

public:
	LinPlayer(QWidget *parent = 0);
	~LinPlayer();

	void DisableAllButton(void);

private slots:
	/**********************************************************************************/
	/* 函数: void ctrl_open(void)                                                     */
	/* 说明：open按钮触发事件处理函数，弹出对话框让用户选择播放文件。并触发play按钮   */
	/* 返回：无                                                                       */
	/* 参数：无                                                                       */
	/**********************************************************************************/
	void ctrl_open(void);

	/**********************************************************************************/
	/* 函数: void ctrl_play(void);                                                    */
	/* 说明：play按钮触发事件处理函数。调用播放函数播放多媒体，若正在播放，则暂停     */
	/* 返回：无                                                                       */
	/* 参数：无                                                                       */
	/**********************************************************************************/
	void ctrl_play(void);

	/**********************************************************************************/
	/* 函数: void setStyleSheet_btn_play(bool bPause);                                */
	/* 说明：设置播放按钮的样式表.                                                    */
	/* 返回：无                                                                       */
	/* 参数：bPause true代表暂停，否则为播放状态                                      */
	/**********************************************************************************/
	void setStyleSheet_btn_play(bool bPause);

	/**********************************************************************************/
	/* 函数: void setStyleSheet_btn_Sound(bool bMute);                                */
	/* 说明：设置声音按钮的样式表.                                                    */
	/* 返回：无                                                                       */
	/* 参数：bMute true代表静音,否则代表有声音.                                       */
	/**********************************************************************************/
	void setStyleSheet_btn_Sound(bool bMute);

	/**********************************************************************************/
	/* 函数: void ctrl_stop(void);                                                    */
	/* 说明：stop按钮触发事件处理函数。调用停止函数停止多媒体并清除相关资源           */
	/* 返回：无                                                                       */
	/* 参数：无                                                                       */
	/**********************************************************************************/
	void ctrl_stop(void);

	/**********************************************************************************/
	/* 函数: void ctrl_previous(void);                                                */
	/* 说明：previous按钮触发事件处理函数。切换到列表中的上一个文件进行播放           */
	/* 返回：无                                                                       */
	/* 参数：无                                                                       */
	/**********************************************************************************/
	void ctrl_previous(void);

	/**********************************************************************************/
	/* 函数: void ctrl_next(void);                                                    */
	/* 说明：previous按钮触发事件处理函数。切换到列表中的下一个文件进行播放           */
	/* 返回：无                                                                       */
	/* 参数：无                                                                       */
	/**********************************************************************************/
	void ctrl_next(void);

	void ctrl_list(void);
	void ctrl_fullscreen(void);

	void ctrl_sound(void);

private:
	void ReadSettings(void);

private:
	Ui::LinPlayerClass ui;
	CPlayCore *player;
	QString strUrl;
};

#endif // LINPLAYER_H
